kafka-client/src/main/java/dev/responsive/kafka/api/ResponsiveGlobalKeyValueBytesStoreSupplier.java

```diff
@@ -0,0 +1,62 @@
+/*
+ * Copyright 2023 Responsive Computing, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package dev.responsive.kafka.api;
+
+import dev.responsive.db.CassandraClient;
+import dev.responsive.kafka.store.ResponsiveGlobalStore;
+import dev.responsive.utils.RemoteMonitor;
+import java.util.concurrent.ScheduledExecutorService;
+import org.apache.kafka.common.utils.Bytes;
+import org.apache.kafka.streams.state.KeyValueBytesStoreSupplier;
+import org.apache.kafka.streams.state.KeyValueStore;
+
+public class ResponsiveGlobalKeyValueBytesStoreSupplier implements KeyValueBytesStoreSupplier {
+
+  private final CassandraClient client;
+  private final String name;
+  private final RemoteMonitor awaitTable;
+
+  public ResponsiveGlobalKeyValueBytesStoreSupplier(
+      final CassandraClient client,
+      final String name,
+      final ScheduledExecutorService executorService
+  ) {
+    this.client = client;
+    this.name = name;
+
+    // we maintain the name without quotes because quotes are
+    // not valid in Kafka topics, but on the other hand they
+    // are necessary to ensure that Cassandra can accept whatever
+    // the client tosses at it
+    awaitTable = client.awaitTable('"' + name + '"', executorService);
```

Comment:
> they are necessary to ensure that Cassandra can accept whatever the client tosses at it

Oh yeah I've been meaning to ask, can you elaborate on why we have to add quotes here? So the table name would be something like `String tableName = "\"store-name\"";`, is that our requirement or Scylla/C*'s? Who does "the client" refer to here?

Reply:
it's a SQL thing - if you don't use quotes then (a) it's not case sensitive and (b) it can't use some characters that are valid in Kafka topics but not in SQL tables. Here's an example. I remember this being necessary for some of my tests, though I'll confirm when I get back from my trip.